### ISSUE TITLE
Investigation of mask R-CNN for removing ripples

### ISSUE THREAD

**Issue by JasonGellis** [alan-turing-institute/Palaeoanalytics#57]
Working from https://github.com/matterport/Mask_RCNN to generate bounding boxes and segmentation masks for each instance (ripples) of an object in the image. It's based on Feature Pyramid Network (FPN) and a ResNet101 backbone.

- [x] Setup pre trained model using COCO data set.
- [x] Image annotation (https://github.com/jsbroks/coco-annotator/wiki/Getting-Started)
- training datasets have been annotated and .json files have been created for test run.
- [x] Train mask R-CNN on own data set
- Script for model has been run, however laptop GPU is not powerful enough.
- Constructiing virtual environment to run on Google Colab Notebooks.
- [ ] Anchor sorting and filtering
- [ ] Refine bounding boxes
- [ ] Generate masks
- [ ] Debugging and visualization (weigthed histograms, tensorboard, layer activation).
- [ ] Finalize

### 1. Setup pretrained model using COCO dataset

(https://github.com/matterport/Mask_RCNN)
Requirements:Python version 3.4 - 3.6. In some cases python packages may need to be downgraded. This is especially the case for Tensorflow. Current requirements below (included in requirements.txt in project folder): 

jupyter
numpy<2.0,>=1.16.0
scipy
Pillow
cython
matplotlib
tensorflow==1.15.0
keras==2.2.4
opencv-python
h5py
imgaug
IPython[all]

### 2. Image annotation. 

https://github.com/jsbroks/coco-annotator/wiki/Getting-Started
COCO-annotator has all the built in tools and applications for R-CNN and Mask R-CNN. 

Currently, the only way to install this is through Docker - https://www.docker.com/
